java/ql/lib/semmle/code/java/security/InsufficientKeySize.qll

```diff
@@ -0,0 +1,197 @@
+/** Provides classes and predicates related to insufficient key sizes in Java. */
+
+private import semmle.code.java.security.Encryption
+private import semmle.code.java.dataflow.DataFlow
+
+/** A source for an insufficient key size. */
+abstract class InsufficientKeySizeSource extends DataFlow::Node {
+  /** Holds if this source has the specified `state`. */
+  predicate hasState(DataFlow::FlowState state) { state instanceof DataFlow::FlowStateEmpty }
+}
+
+/** A sink for an insufficient key size. */
+abstract class InsufficientKeySizeSink extends DataFlow::Node {
+  /** Holds if this sink has the specified `state`. */
+  predicate hasState(DataFlow::FlowState state) { state instanceof DataFlow::FlowStateEmpty }
+}
+
+private module Asymmetric {
+  private module NonEllipticCurve {
+    /** A source for an insufficient key size used in RSA, DSA, and DH algorithms. */
+    private class AsymmetricNonEcSource extends InsufficientKeySizeSource {
```

Comment:
Now that we're using modules, do you think these classes and predicates need the `[Asymmetric[Non]Ec|Symmetric]-` prefixes? I'd slightly prefer shorter versions that'd make sense based on the modules they're in (`Source`, `Sink`, `Spec`, `getMinKeySize`, etc...). They're private anyway, but if they were referenced externally, the module syntax would provide the context IMHO (e.g. `Asymmetric::NonEllipticCurve::Source`).

But again, no strong opinion, so feel free to disregard this suggestion.

Reply:
Updated in https://github.com/github/codeql/pull/10785/commits/1bfdfc954bb00c6b6c160dbfdf35472556d49614.